worker/crates/action-universal/src/attribute/length_calculator.rs

```diff
@@ -0,0 +1,56 @@
+use serde::{Deserialize, Serialize};
+
+use reearth_flow_action::{
+    error::Error, ActionContext, ActionDataframe, ActionResult, AsyncAction, Attribute,
+    AttributeValue, Dataframe, Feature, DEFAULT_PORT,
+};
+use serde_json::Number;
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct AttributeStringLengthCalculator {
+    source_attribute: String,
+    string_length_attribute: String,
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "AttributeStringLengthCalculator")]
+impl AsyncAction for AttributeStringLengthCalculator {
+    async fn run(&self, _ctx: ActionContext, inputs: ActionDataframe) -> ActionResult {
+        let dataframe = inputs
+            .get(&DEFAULT_PORT)
+            .ok_or(Error::input("no default"))?
+            .clone();
+        let lens = dataframe
+            .features
+            .into_iter()
+            .map(|x| {
+                if let Some(AttributeValue::String(src)) =
+                    x.attributes.get(&Attribute::new(&self.source_attribute))
+                {
+                    Feature::new_with_attributes(
```

Comment:
![must](https://img.shields.io/badge/review-must-critical.svg)
Do not delete existing data, but keep it.

Reply:
I has corrected.